internal/controller/drplacementcontrol_controller.go

```diff
@@ -565,40 +565,36 @@ func GetDRClusters(ctx context.Context, client client.Client, drPolicy *rmn.DRPo
 }
 
 // updateObjectMetadata updates drpc labels, annotations and finalizer, and also updates placementObj finalizer
-func (r DRPlacementControlReconciler) updateObjectMetadata(ctx context.Context,
-	drpc *rmn.DRPlacementControl, placementObj client.Object, log logr.Logger,
-) error {
-	var update bool
+func (r DRPlacementControlReconciler) updateObjectMetadata(
+	drpc *rmn.DRPlacementControl, placementObj client.Object,
+) (drpcModified bool, placementModified bool, err error) {
+	// Explicit initialization
+	drpcModified = false
+	placementModified = false
+
+	if rmnutil.AddLabel(drpc, rmnutil.OCMBackupLabelKey, rmnutil.OCMBackupLabelValue) {
+		drpcModified = true
+	}
 
-	update = rmnutil.AddLabel(drpc, rmnutil.OCMBackupLabelKey, rmnutil.OCMBackupLabelValue)
-	update = rmnutil.AddFinalizer(drpc, DRPCFinalizer) || update
+	added := rmnutil.AddFinalizer(drpc, DRPCFinalizer)
+	if added {
```

Comment:
The "added" variable does not strictly need to be declared. You can inline the function call directly into the if statement